lib/ecstatic.js

```diff
@@ -330,6 +339,19 @@ var ecstatic = module.exports = function (dir, options) {
 ecstatic.version = version;
 ecstatic.showDir = showDir;
 
+function hasGzipId12(gzipped, cb) {
+  var stream = fs.createReadStream(gzipped, { start: 0, end: 1, autoClose: true }),
+      isGzip = false;
+
+  stream.on("close", function () {
+    cb(isGzip);
+  });
+
+  stream.on("data", function (chunk) {
```

Comment:
As a matter of interface, all callbacks must be of the form `(err, result)` even if that error is always null. I don't really like the idea of swallowing errors, but buy the argument that the right behavior overall is to swallow errors and callback false. However, last I checked an emitted error event on a stream will throw if you don't explicitly add an error handler, so even if you callback null, false you still need to listen for that event.

I confirmed the buffer length check is unnecessary. Weird that buffers allow for that, but ok.